### DISCUSSION TITLE
Migrate window positions from old to new version

### DISCUSSION THREAD

**Discussion by daxliniere** [winder/Universal-G-Code-Sender#2052] [Help] [answered]
My old version is snapshot 2022-09-30 and I have just installed the newest snapshot (2022-11-04).
I have tried overwriting all files in **C:\Users\dax\.ugsplatform\2.0-SNAPSHOT** with all files in **C:\Users\dax\.ugsplatform\2.0.11** and vice-versa (I took backups, duh ;) ), but neither seems to make 2022-11-04 display my panes like in 2022-09-30.

Where is this stored, please?

![image](https://user-images.githubusercontent.com/8109020/201788140-0914e43d-f3ea-4f1e-a8d6-1559c5dea5f5.png)
![image](https://user-images.githubusercontent.com/8109020/201788159-37e98d40-19f7-464f-941d-46e0a4ada1dc.png)

**Comment by breiler** [comment]
The settings has been a mess and it hasn't been working properly on some platforms. In an attempt to fix this I tried changing how the settings are stored in #2038 which will be available in the next version 2.0.13 (it is currently available in the nightly build).

If you have backups of the settings under `C:\Users\dax\.ugsplatform` you can remove everything under that directory. Then copy the contents from one of the old settings directory: `C:\Users\dax.ugsplatform\2.0.11\dev\`.

The directory `C:\Users\dax\.ugsplatform` should now contain the directories `config` and `var`.

**Comment by daxliniere** [comment]
So I think this is not correct, then?
![image](https://user-images.githubusercontent.com/8109020/202052630-3e8bacdc-a605-4aba-863d-9a3b31b41abc.png)
It seems the latest version hasn't created config and var where it should have.

Should I just move the most recent config and var directories to C:\Users\dax\.ugsplatform ?

**Reply by daxliniere** [reply]
@breiler, should I move config and var from \2.0-SNAPSHOT\dev into this directory?

**Comment by daxliniere** [comment]
Any news on this, Joacim? Where should I put the files so 2022-11-04 and later will read them, please?

**Comment by breiler** [comment]
I was about to setup a Windows test machine to figure this out. But JFrog has stopped working again so I started working on a new build script using github actions but didn't have time to finish it.

**Reply by daxliniere** [reply]
>...I started working on a new build script using github actions but didn't have time to finish it.
No worriess, mate, never any pressure from me, I know you're a busy man. I was just curious how it was all going. 😊

**Comment by breiler** [comment, marked as answer]
I have made an update in the nightly build, fixing a problem with windows not using the correct path for cache files.

The settings for UGS will now be stored under `c:/users/<your username>/AppData/Roaming/ugsplatform/`.

**Comment by daxliniere** [comment]
Thanks Joacim. When I copied all files from the old settings folders to the new ones, the layout now contained all the windows, though their positions were slightly rearranged. Certainly not a problem, I'm sure that once I rearrange them, they'll stay in the new positions.

Nice fix, mate. Cheers.
